src/analysis/retail/monk/mistweaver/modules/dragonflight/tier/T29MWTier.tsx

```diff
@@ -0,0 +1,200 @@
+import Analyzer, { Options, SELECTED_PLAYER } from 'parser/core/Analyzer';
+import { calculateEffectiveHealing } from 'parser/core/EventCalculateLib';
+import { formatDuration, formatNumber } from 'common/format';
+import SPELLS from 'common/SPELLS';
+import { TALENTS_MONK } from 'common/TALENTS';
+import { TIERS } from 'game/TIERS';
+import Events, { HealEvent, ApplyBuffEvent, HasRelatedEvent } from 'parser/core/Events';
+import BoringValueText from 'parser/ui/BoringValueText';
+import ItemHealingDone from 'parser/ui/ItemHealingDone';
+import Statistic from 'parser/ui/Statistic';
+import STATISTIC_CATEGORY from 'parser/ui/STATISTIC_CATEGORY';
+import STATISTIC_ORDER from 'parser/ui/STATISTIC_ORDER';
+import HotTracker from 'parser/shared/modules/HotTracker';
+import HotTrackerMW from '../../core/HotTrackerMW';
+import SpellLink from 'interface/SpellLink';
+
+const TWO_PIECE_BONUS = 0.1;
+const FOUR_PIECE_BONUS = 0.1;
+const FOUR_PIECE_EXTENSION = 1000;
+const ATTRIBUTION_PREFIX = '4 Piece extension';
+
+const TWO_PIECE_SPELLS = [
+  SPELLS.ESSENCE_FONT_BUFF,
+  SPELLS.FAELINE_STOMP_ESSENCE_FONT,
+  TALENTS_MONK.ENVELOPING_MIST_TALENT,
+  SPELLS.VIVIFY,
+];
+
+const FOUR_PIECE_SPELLS = [
+  SPELLS.FAELINE_STOMP_ESSENCE_FONT,
+  SPELLS.ESSENCE_FONT_BUFF,
+  SPELLS.VIVIFY,
+];
+
+const EXTENSION_ATTRIB = HotTracker.getNewAttribution(ATTRIBUTION_PREFIX);
+
+class T29TierSet extends Analyzer {
+  static dependencies = {
+    hotTracker: HotTrackerMW,
+  };
+  hotTracker!: HotTrackerMW;
+  has2Piece: boolean = true;
+  has4Piece: boolean = true;
+  numExtensions: number = 0;
+  totalExtensionDuration: number = 0;
+  twoPieceHealing: number = 0;
+  fourPieceHealingFromBuff: number = 0;
+  extraRemHealing: number = 0;
+  extraVivHealing: number = 0;
+
+  constructor(options: Options) {
+    super(options);
+    this.has2Piece = this.selectedCombatant.has2PieceByTier(TIERS.T29);
+    this.has4Piece = this.selectedCombatant.has4PieceByTier(TIERS.T29);
+    this.active = this.has2Piece || this.has4Piece;
```

Comment:
`|| this.has4Piece` is unnecessary because of your if statement on the event listeners below